src/agentscope/server/servicer.py

```diff
@@ -109,6 +117,8 @@ def __init__(
         self.agent_id_lock = threading.Lock()
```

Comment:
Since agent id is a uuid with 128bits, why we use a threading.Lock here?

Is it for the pooling of agent id or agent id itself?

Reply:
Prevent data structure errors that may result from simultaneous reading and writing